src/osd/PG.cc

```diff
@@ -4233,7 +4235,6 @@ void PG::scrub_process_inconsistent()
     dout(2) << ss.str() << dendl;
     osd->clog->error(ss);
     if (repair) {
-      state_clear(PG_STATE_CLEAN);
```

Comment:
In PG::scrub, if !is_clean(), it will terminate scrub. 
At first, i want to do chunk_repair which if there is missing or inconsistent object.This is the same as you said add a new state. But for repair, it use DoRecovery. This will clean scrub state. 
I think we can only add missing/inconsistent object pg_log/missing_loc which make the client know this. 
Could you give some suggestion?